src/matter/cluster/LevelControlCluster.ts

```diff
@@ -0,0 +1,179 @@
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.5.8 */
+export const OptionsBitmap = {
+    /** Command execution control - see 1.6.5.8.1. */
+    ExecuteIfOff: BitFlag(0), // TODO default: true.
+
+    /** level effect on color - see 1.6.5.8.2 */
+    CoupleColorTempToLevel: BitFlag(1),
+};
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.1 */
+const  MoveToLevelCommand = TlvObject({
+    level: TlvField(0, TlvUInt8.bound({min:0, max:254})),
+    transitionTime: TlvField(1, TlvNullable(TlvUInt16)),
+    optionsMask: TlvField(2, TlvBitmap(TlvUInt8, OptionsBitmap)),
+    optionsOverride: TlvField(3, TlvBitmap(TlvUInt8, OptionsBitmap)), // TODO: 0 Default - all optionsOverride
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.2 */
+const  MoveCommand = TlvObject({
+    moveMode: TlvField(0, TlvEnum<MoveMode>()),
+    rate: TlvField(1, TlvNullable(TlvUInt8)),
+    optionsMask: TlvField(2, TlvBitmap(TlvUInt8, OptionsBitmap)),
+    optionsOverride: TlvField(3, TlvBitmap(TlvUInt8, OptionsBitmap)),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.3 */
+const  StepCommand = TlvObject({
+    stepMode: TlvField(0, TlvEnum<StepMode>()),
+    stepSize: TlvField(1, TlvUInt8),
+    transitionTime: TlvField(2, TlvNullable(TlvUInt16)),
+    optionsMask: TlvField(3, TlvBitmap(TlvUInt8, OptionsBitmap)),
+    optionsOverride: TlvField(4, TlvBitmap(TlvUInt8, OptionsBitmap)),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.4 */
+const  StopCommand = TlvObject({
+    optionsMask: TlvField(0, TlvBitmap(TlvUInt8, OptionsBitmap)),
+    optionsOverride: TlvField(1, TlvBitmap(TlvUInt8, OptionsBitmap)),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 1.6.6.5  */
+const MoveToClosestFrequencyCommand = TlvObject({
+    frequency: TlvField(0, TlvUInt16 ), // FIXME - Default 0, how to set TlvField default?
+})
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.4 */
+const features = {
+    /** Dependency with the On/Off cluster */
+    OnOff: BitFlag(0), // TODO default: true.
+
+    /** Behavior that supports lighting applications */
+    Lighting: BitFlag(1),
+
+    /** Supports frequency attributes and behavior. */
+    Frequency: BitFlag(2)
+};
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6.2.1 */
+const enum MoveMode {
+    up = 0x0,
+    down = 0x1,
+};
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.6.3 */
+const enum StepMode {
+    up = 0x0,
+    down = 0x1,
+};
+
+ /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.6.5 */
+const attributes = {
+    /** the current level of this device */
+    currentLevel: Attribute(0x0,  TlvNullable(TlvUInt8), {default: null}), 
+
+    /** time until the current command is complete in 1/10ths of sec */
+    remainingTime: OptionalAttribute(0x01, TlvUInt16, {default: 0}),
+
+    /** the minimum value of CurrentLevel  */
+    minLevel: OptionalAttribute(0x02, TlvUInt8.bound({ min: 0})),
```

Comment:
This is a TlvUInt8 so the min is already 0 (unsigned int 8 bit), so no need to redeclare that.

Sam below (all unsigned int are already internally constraints to be >= 0)